src/patternfly/components/Button/button.scss

```diff
@@ -127,6 +128,10 @@
     margin-right: var(--pf-c-button__icon--MarginRight);
   }
 
+  .pf-c-button__text + .pf-c-button__icon {
+    margin-right: 0;
+    margin-left: var(--pf-c-button__icon--MarginLeft);
+  }
```

Comment:
Instead of `.pf-c-button__text + . pf-c-button__icon`, what about:

```
> * {
  margin-right: var(--pf-c-button--item--MarginRight);
}

> :last-child {
  --pf-c-button--item--MarginRight: 0;
}
```

**Why?** 
- Maintains base specificity.
- Don't have to worry about swapping `margin-right/left`.
- Simpler

Reply:
That would change the natural spacing if people have their own elements in the button, like `<button><span>word</span> <span>word</span></button>`

Reply:
Right, I would expect that.

What about something like this when it comes along?
![Screen Shot 2020-01-09 at 11 30 43 AM](https://user-images.githubusercontent.com/5385435/72085540-817a8500-32d3-11ea-8410-1afca8d77cf5.png)

Reply:
@mcoker This solution works, but I think it's too specific/needs to be more flexible and we'll be revisiting in the future.

Reply:
> Right, I would expect that.

I wouldn't expect a space difference between `<button>word word</button>` and `<button><span>word</span> <span>word</span></button>` though, and we never know what a person will put in the button. I agree we'll want to support space between other things in the future for sure, but as long as users can put anything they want in the button, I don't think a margin on every direct child is the best approach.

Reply:
Ship it